sqlmesh/core/engine_adapter/logical_merge.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import typing as t
+
+from sqlglot import exp, parse_one
+
+from sqlmesh.core.engine_adapter.base import EngineAdapter, TransactionType
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core._typing import TableName
+    from sqlmesh.core.engine_adapter._typing import QueryOrDF
+
+    class Base(EngineAdapter):
+        ...
+
+else:
+
+    class Base:
+        ...
+
+
+class LogicalMergeMixin(Base):
+    def merge(
+        self,
+        target_table: TableName,
+        source_table: QueryOrDF,
+        columns_to_types: t.Dict[str, exp.DataType],
+        unique_key: t.Sequence[str],
+    ) -> None:
+        temp_table = self._get_temp_table(target_table)
+        unique_exp: exp.Expression = parse_one("|| '__SQLMESH_DELIM__' ||".join(unique_key))
```

Comment:
Is it ok to rely on implicit casting of the columns to string in order to concat?

Reply:
This is how dbt does it. Is there a particular case you are worried about?

Reply:
Ok then that implicit cast must be widely supported across engines.